### ISSUE TITLE
Out of memory exceptions

### ISSUE THREAD

**Issue by czarifis** [Rudrabha/Wav2Lip#106]
When applying inference on videos with higher resolution I often get OOM exceptions. In some cases the memory utilization spikes to 32GB or more.

I believe this either happens when trying to read input frames (probably while detecting the face) or when applying the lip movement. Is there any way to avoid this by doing some sort of batching? Perhaps by cutting the video & audio to a few seconds, applying the inference and then concatenating? 

Let me know what you think!

Thanks everyone! Awesome work!

**Comment by prajwalkr** [comment]
> In some cases the memory utilization spikes to 32GB or more.

You must be taking an inference on a very long video then?

**Comment by czarifis** [comment]
It's about 1 minute at 1080p.

**Comment by prajwalkr** [comment]
You can try to write a script that chunks the video/audio and calls this script, gets the result and combines them back.

**Comment by czarifis** [comment]
For anyone that might be having the same issue, it looks like the problem has to do with the resolution of the video and not so much with the length. I tried splitting the video into 1-second slices but unfortunately I was still getting out of memory errors for 1080p videos.

What worked for me was cropping the video so that the footage is focused on the face. With this approach, we managed to lowered the size of each frame without lowering the quality of the footage.

If you're using ffmpeg this is the command I used:
`ffmpeg -i input.mp4 -filter:v "crop=1000:700:200:0" out.mp4`

For more info on how to use the crop filter in ffmpeg, refer to this tutorial: https://www.linuxuprising.com/2020/01/ffmpeg-how-to-crop-videos-with-examples.html
